tinygrad/helpers.py

```diff
@@ -96,16 +96,19 @@ def fields() -> Dict[str, DType]: return DTYPES_DICT
   float32: Final[DType] = DType(4, 4, "float", np.float32)
   float = float32
   int8: Final[DType] = DType(0, 1, "char", np.int8)
-  int32: Final[DType] = DType(1, 4, "int", np.int32)
-  int64: Final[DType] = DType(2, 8, "long", np.int64)
+  int16: Final[DType] = DType(1, 2, "short", np.int16)
+  int32: Final[DType] = DType(2, 4, "int", np.int32)
+  int64: Final[DType] = DType(3, 8, "long", np.int64)
   uint8: Final[DType] = DType(0, 1, "unsigned char", np.uint8)
-  uint32: Final[DType] = DType(1, 4, "unsigned int", np.uint32)
-  uint64: Final[DType] = DType(2, 8, "unsigned long", np.uint64)
+  uint16: Final[DType] = DType(1, 2, "unsigned short", np.uint16)
+  uint32: Final[DType] = DType(2, 4, "unsigned int", np.uint32)
+  uint64: Final[DType] = DType(3, 8, "unsigned long", np.uint64)
 
   # NOTE: bfloat16 isn't supported in numpy
   bfloat16: Final[DType] = DType(0, 2, "__bf16", None)
 
   # NOTE: these are internal dtypes, should probably check for that
+  _bfloat16: Final[DType] = DType(0, 2, "bfloat16", None)
```

Comment:
Why can't the `bfloat16` dtype above be used?

Reply:
Because then it works only on LLVM and no other backend.

Reply:
The idea with it was to load the weights to LLVM cast to fp16, then move them to whatever device they will be used on.

Reply:
That sounds good. However, I have no idea how to do this because of my superficial knowledge of the tinygrad architecture. Feel free to take over this PR and update it to match your expectations.

Reply:
If we're using pytorch for the loading anyway, why not just do `.half()` after loading?
(I confirmed `x.half()` gave the same result as `x.float().half()`)

Reply:
Ah nevermind, I see we're using the tinygrad torch loader. My bad.

Reply:
> The idea with it was to load the weights to LLVM cast to fp16

@wozeparrot can you provide some hints how to do this so I can rework the PR?

Reply:
You should be able to do something like: `x.to("llvm").float().realize().to(Device.DEFAULT).realize()` where `x` is the loaded weight

Reply:
Hm, I tried the following:

``` python
    if storage[1] == dtypes.bfloat16:
      ret = t[byte_offset:byte_offset+prod(size)].to("LLVM").half().realize().to(t.device).realize()
    else:
      ret = t[byte_offset:byte_offset+prod(size)].cast(storage[1])
```

and it seems it modified weights on my disk :-/ Redownloading LLaMA-2-7b weights again ...

Reply:
oh yea can't do t.device because the weights are mmapped from disk, so the tensor's device is disk. So moving the weights to that device is the same as writing to disk.

Reply:
> so the tensor's device is disk

Learned the hard way, but it makes sense! :-)

Reply:
Pushed bf167b8db1fd5f3e5a2125b1d35f0abd9262ba0c which converts BFloat16 to Float16 via LLVM, but I am unable to get meaningful results (Stacy keeps talking jibberish).

I am once again out of ideas.

Reply:
maybe try something like:

`ret = t[byte_offset:byte_offset+prod(size)].contiguous().realize().to("LLVM").realize().half().realize().to(Device.DEFAULT).realize()`

Reply:
@wozeparrot nope, that didn't help